### PR TITLE
Add approval steps

### PR DESCRIPTION
**Pull request by osterman**
## what
* Add approval steps

## why
* Demonstrate flow control

## demo

image
image

### CODE REVIEW

**Review by aknysh — APPROVED** [review]

File: codefresh/destroy.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ stages:
 
 steps:
   main_clone:
```
Comment (aknysh) [inline codefresh/destroy.yaml:7]:
```suggestion
  ask_for_permission:
```

File: codefresh/destroy.yaml [hunk 2]
```diff
@@ -15,3 +19,8 @@ steps:
     - "kubectl config use-context ${{KUBE_CONTEXT}}"
     - "make helm/delete/namespace"
     - "make helm/delete/namespace/empty"
+    when:
+      steps:
+      - name: main_clone
```
Comment (aknysh) [inline codefresh/destroy.yaml:24]:
```suggestion
      - name: ask_for_permission
```

**Review by aknysh — COMMENTED** [review]
see comments

File: codefresh/destroy.yaml (hunk 1 above)
Comment (osterman) [reply codefresh/destroy.yaml:7]: First step in a codefresh pipeline should be called `main_clone`, otherwise codefresh adds it's own `main_clone` step

**Review by aknysh — CHANGES_REQUESTED** [review]
a few nitpicks

**Review by aknysh — APPROVED** [review]

**Pull request merged**